libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/map/MapViewBinder.kt

```diff
@@ -21,47 +18,57 @@ import com.mapbox.navigation.dropin.map.longpress.RoutePreviewLongPressMapCompon
 import com.mapbox.navigation.dropin.map.marker.MapMarkersComponent
 import com.mapbox.navigation.dropin.map.scalebar.ScalebarComponent
 import com.mapbox.navigation.dropin.navigationview.NavigationViewContext
-import com.mapbox.navigation.ui.app.internal.Store
 import com.mapbox.navigation.ui.app.internal.navigation.NavigationState
-import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewAction
-import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewState
-import com.mapbox.navigation.ui.app.internal.routefetch.RoutesAction
 import com.mapbox.navigation.ui.base.lifecycle.UIBinder
 import com.mapbox.navigation.ui.maps.internal.ui.LocationComponent
 import com.mapbox.navigation.ui.maps.internal.ui.LocationPuckComponent
 import com.mapbox.navigation.ui.maps.internal.ui.RouteArrowComponent
 import com.mapbox.navigation.ui.maps.internal.ui.RouteLineComponent
-import com.mapbox.navigation.ui.maps.internal.ui.RouteLineComponentContract
 import com.mapbox.navigation.ui.maps.route.arrow.model.RouteArrowOptions
 import com.mapbox.navigation.ui.maps.route.line.model.MapboxRouteLineOptions
-import kotlinx.coroutines.FlowPreview
-import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.combine
 
 @ExperimentalPreviewMapboxNavigationAPI
-@FlowPreview
-internal class MapBinder(
-    private val context: NavigationViewContext,
-    private val binding: MapboxNavigationViewLayoutBinding,
-    private val mapView: MapView
-) : UIBinder {
-
-    init {
-        mapView.compass.enabled = false
+abstract class MapViewBinder : UIBinder {
+
+    companion object {
+        /**
+         * Default Info Panel Binder.
+         */
+        fun defaultBinder(): MapViewBinder = MapboxMapViewBinder()
+    }
+
+    internal lateinit var context: NavigationViewContext
+    internal lateinit var navigationViewBinding: MapboxNavigationViewLayoutBinding
+
+    private val mapboxMapObservers = mutableSetOf<MapboxMapObserver>()
+
+    internal fun registerMapboxMapObserver(observer: MapboxMapObserver) {
+        mapboxMapObservers.add(observer)
     }
 
-    private val scalebarComponent = ScalebarComponent(
-        mapView,
-        context.styles.mapScalebarParams,
-        context.systemBarsInsets
-    )
+    abstract fun getMapView(viewGroup: ViewGroup): MapView
+
+    @MapStyleLoadPolicy.MapLoadStylePolicy
+    open fun getMapStyleLoadPolicy(): Int = MapStyleLoadPolicy.NEVER
+
+    internal open fun addMapViewToLayout(mapView: MapView, viewGroup: ViewGroup) {
+        viewGroup.removeAllViews()
+        viewGroup.addView(mapView)
+    }
 
-    private val store = context.store
+    open fun onMapViewReady(mapView: MapView) { }
```

Comment:
```suggestion
    protected open fun onMapViewReady(mapView: MapView) = Unit
```
Since subclasses are allowed to override this method and only this class will call it, let's reduce its visibility to `protected`.

Also, since this method is visible, let's add KDoc for it.